[BUG] Beta 2.13.0 RC1: Microsoft Business OAuth Token too long for database

### Description

When connecting a Microsoft 365 business account with the recently added OAuth feature, the token can not be stored in the db due to length constraints of 2048 characters.

### Steps to reproduce

1. Set up OAuth Config variables for Office365
2. Click the OAuth button in mail settings (paperless)
3. Accept the OAuth Sign-In
4. Get error 500

### Webserver logs

```bash
paperless-db-1         | 2024-10-14 13:54:04.157 UTC [184] ERROR:  value too long for type character varying(2048)
paperless-db-1         | 2024-10-14 13:54:04.157 UTC [184] STATEMENT:  INSERT INTO "paperless_mail_mailaccount" ("owner_id", "name", "imap_server", "imap_port", "imap_security", "username", "password", "is_token", "character_set", "account_type", "refresh_token", "expiration") VALUES (NULL, 'Outlook OAuth 2024-10-14 13:54:03.697806+00:00', 'outlook.office365.com', 993, 2, '', '*deleted token for obvious reasons*', true, 'UTF-8', 3, '*deleted token for obvious reasons*', '2024-10-14 15:10:07.154097+00:00'::timestamptz) RETURNING "paperless_mail_mailaccount"."id"
paperless-webserver-1  | [2024-10-14 15:54:04,158] [ERROR] [django.request] Internal Server Error: /api/oauth/callback/
```


### Browser logs

_No response_

### Paperless-ngx version

2.13.0-beta-rc1

### Host OS

Synology DSM 7.2

### Installation method

Docker - official image

### System status

_No response_

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/paperless_mail/migrations/0027_mailaccount_expiration_mailaccount_account_type_and_more.py
+++ b/src/paperless_mail/migrations/0027_mailaccount_expiration_mailaccount_account_type_and_more.py
@@ -10,6 +10,11 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
+        migrations.AlterField(
+            model_name="mailaccount",
+            name="password",
+            field=models.CharField(max_length=3072, verbose_name="password"),
+        ),
         migrations.AddField(
             model_name="mailaccount",
             name="expiration",

--- a/src/paperless_mail/migrations/0027_mailaccount_expiration_mailaccount_account_type_and_more.py
+++ b/src/paperless_mail/migrations/0027_mailaccount_expiration_mailaccount_account_type_and_more.py
@@ -40,7 +40,7 @@ class Migration(migrations.Migration):
             field=models.CharField(
                 blank=True,
                 help_text="The refresh token to use for token authentication e.g. with oauth2.",
-                max_length=2048,
+                max_length=3072,
                 null=True,
                 verbose_name="refresh token",
             ),

--- a/src/paperless_mail/models.py
+++ b/src/paperless_mail/models.py
@@ -42,7 +42,7 @@ class MailAccount(document_models.ModelWithOwner):
 
     username = models.CharField(_("username"), max_length=256)
 
-    password = models.CharField(_("password"), max_length=2048)
+    password = models.CharField(_("password"), max_length=3072)
 
     is_token = models.BooleanField(_("Is token authentication"), default=False)
 

--- a/src/paperless_mail/models.py
+++ b/src/paperless_mail/models.py
@@ -64,7 +64,7 @@ class MailAccount(document_models.ModelWithOwner):
 
     refresh_token = models.CharField(
         _("refresh token"),
-        max_length=2048,
+        max_length=3072,
         blank=True,
         null=True,
         help_text=_(

